src/robot/libraries/Process.py

```diff
@@ -407,7 +407,10 @@ def wait_for_process(self, handle=None, timeout=None, on_timeout='continue'):
         given in
         [http://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#time-format|
         various time formats] supported by Robot Framework, for example, ``42``,
-        ``42 s``, or ``1 minute 30 seconds``.
+        ``42 s``, or ``1 minute 30 seconds``. ``NONE``, ``0``, or ``0s`` values
+        set timeout to be infinite (as shown
+        [https://robotframework.org/robotframework/latest/RobotFrameworkUserGuide.html#test-case-timeout|here])
+        ``-int`` values are ignored (i.e. set timeout to be infinite).
```

Comment:
This somehow sounds overly complicated and referring to the User Guide is not necessary. Something like this could work better:

> ``NONE`` (case-insensitively) is same as not specifying timeout at all. Starting from Robot Framework 3.2 also zero and negative timeout values are ignored.